### PR TITLE
find: Add -readable, -writable & -executable options

### PR DESCRIPTION
**Pull request by ricardobranco777**
Add -readable, -writable & -executable options for GNU find compatibility.  These functions use access() to test for readable/writable/executable access by the current user.

References:
- [GNU find manual](https://manpages.ubuntu.com/manpages/jammy/en/man1/find.1.html)

### CODE REVIEW

File: usr.bin/find/find.1 [hunk 1]
```diff
@@ -465,6 +467,11 @@ if at least one invocation of
 returns a non-zero exit status,
 .Nm
 will return a non-zero exit status.
+.It Ic -executable
+Matches files which are executable by the current user. This test makes use of the
```
Comment (igalic) [inline usr.bin/find/find.1:471]: start new sentences on new lines

File: usr.bin/find/find.1 [hunk 2]
```diff
@@ -829,6 +836,11 @@ option was specified.
 Causes
 .Nm
 to terminate immediately.
+.It Ic -readable
```
Comment (igalic) [inline usr.bin/find/find.1:839]: start new sentences on new lines

File: usr.bin/find/find.1 [hunk 3]
```diff
@@ -925,6 +937,11 @@ is treated as a user ID.
 The same thing as
 .Ic -path ,
 for GNU find compatibility.
+.It Ic -writable
+Matches files which are writable by the current user. This test makes use of the
```
Comment (igalic) [inline usr.bin/find/find.1:941]: start new sentences on new lines

File: usr.bin/find/option.c [hunk 4]
```diff
@@ -75,6 +75,7 @@ static OPTION const options[] = {
 	{ "-empty",	c_empty,	f_empty,	0 },
 	{ "-exec",	c_exec,		f_exec,		0 },
 	{ "-execdir",	c_exec,		f_exec,		F_EXECDIR },
+	{ "-executable",c_simple,	f_executable,	0 },
```
Comment (igalic) [inline usr.bin/find/option.c:78]: space after comma

**Review by igalic — CHANGES_REQUESTED** [review]
a few nits i noticed on a first read thru

File: usr.bin/find/find.1 (hunk 1 above)
Comment (ricardobranco777) [reply usr.bin/find/find.1]: Fixed.

File: usr.bin/find/find.1 [hunk 5]
```diff
@@ -28,7 +28,9 @@
 .\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 .\" SUCH DAMAGE.
 .\"
-.Dd January 15, 2024
+.\"	@(#)find.1	8.7 (Berkeley) 5/9/95
```
Comment (bsdjhb) [inline usr.bin/find/find.1:31]: Mis-merge with changes in main to remove old SCCS IDs?

**Review by bsdjhb — APPROVED** [review]

**Comment by bsdjhb** [comment]
@bsdimp The style violations from the checker are false positives.

**Review by igalic — APPROVED** [review]
👍 (with @bsdjhb's complaint addressed)

**Comment by bsdjhb** [comment]
Merged in e4b646ce16105efc943bc8ac2242a2220dfe18d8, thanks!

**Pull request closed without merging**